services/notification/notifiers/comment/conditions.py

```diff
@@ -107,12 +113,113 @@ def check_condition(
 class HasEnoughRequiredChanges(AsyncNotifyCondition):
     failure_explanation = "changes_required"
 
+    async def _check_unexpected_changes(comparison: ComparisonProxy) -> bool:
+        """Returns a bool that indicates wether there are unexpected changes"""
+        changes = await comparison.get_changes()
+        if changes:
+            return True
+        return False
+
+    async def _check_coverage_change(comparison: ComparisonProxy) -> bool:
+        """Returns a bool that indicates wether there is any change in coverage"""
+        diff = await comparison.get_diff()
+        res = comparison.head.report.calculate_diff(diff)
+        if res is not None and res["general"].lines > 0:
+            return True
+        return False
+
+    async def _check_any_change(comparison: ComparisonProxy) -> bool:
+        unexpected_changes = await HasEnoughRequiredChanges._check_unexpected_changes(
+            comparison
+        )
+        coverage_changes = await HasEnoughRequiredChanges._check_coverage_change(
+            comparison
+        )
+        return unexpected_changes or coverage_changes
+
+    async def _check_coverage_drop(comparison: ComparisonProxy) -> bool:
+        no_head_coverage = comparison.head.report.totals.coverage is None
+        no_base_report = comparison.project_coverage_base.report is None
+        no_base_coverage = (
+            no_base_report
+            or comparison.project_coverage_base.report.totals.coverage is None
+        )
+        if no_head_coverage or no_base_coverage:
+            # We don't know if there was a coverage drop, because we can't compare BASE and HEAD (missing some info)
+            # But we default to showing the comment. It might have info for the user about _what_ info we are missing
+            return True
+        print(comparison.head.report.totals.coverage)
```

Comment:
is this leftover from testing or do you want to keep it?

Reply:
absolutely left over, good catch 👍